ortools/linear_solver/xpress_interface.cc

```diff
@@ -1625,6 +1678,21 @@ MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
   // Set the hint (if any)
   this->AddSolutionHintToOptimizer();
 
+  // Add opt node callback to optimizer. We have to do this here (just before
+  // solve) to make sure the variables are fully initialized
+  XpressMPCallbackContext* xpress_context;
+  MPCallbackWithXpressContext mp_callback_with_context;
+  if (callback_ != nullptr) {
+    xpress_context = new XpressMPCallbackContext(
```

Comment:
Where is this deleted? Should we call `XPRSremovecboptnode()` and delete this right after `XPRSoptimize()` returns? At the moment it looks we are leaking memory for each solve in case there is a callback registered.

Reply:
I will replace it with a unique pointer